incubator/hnc/api/v1alpha1/hierarchy_types.go

```diff
@@ -103,6 +104,9 @@ type Condition struct {
 	// - "OBJECT_OVERRIDDEN": an object in this namespace has been overridden from its parent and will no longer be updated
 	//
 	// - "OBJECT_DESCENDANT_OVERRIDDEN": an object in this namespace is no longer being propagated because a propagated copy has been modified
+	//
+	// - "OBJECT_INTERNAL": The object has a finalizer set, and should not be propagated.
```

Comment:
@yiqigao217 , what's the rationale behind this name? I'm not sure what "internal" means. Maybe something like OBJECT_METADATA with a description "This object has metadata that prevents it from being propagated to descendant namespaces" and the `msg` field will identify the actual problems?

Reply:
SGTM. Yes, metadata is better.